### ISSUE TITLE
Import/Export Refactor

### ISSUE THREAD

**Issue by vabene1111** [TandoorRecipes/recipes#894]
just collecting things for a future rework

- [x] move to new vue frontend
- [x] one central location to manage importer types
- [x] at that one central location also store some documentation to show the user when selecting a recipe
- [x] some kind of bulk feature #702
- [x] extra tab / documentation for bookmarklet importer
- [x] review if splitting into seperate compontents for the different tabs makes sense
- [x] import servings text #1465
- [x] ~~show list of unimported bookmarklet imports on the bookmarklet tab~~ (makes no sense)
- [x] ~~implement space recipe limiting for both url and application importer~~ (later)
- [x] deprecate all unsused old functions
    - [x] import_url function in data.py
    - [x] url_import.hmtl (and related libraries)
- [x] preference for keyword import #1165
- [x] multi recipe import properly show progress and results afterwards
- [x] add missing docs entries for integration types

for url import
- [x] ~~keep a lit of recent imports on the url page when an import is canceled restarted ~~(not needed)
- [x] automatically load on pasting of url
- [x] allow deciding if steps should be split or merged -> recipe scrapers combines list elements automatically, either try to get an option merged upstream, override or build nice frontend for easy splitting
- [x] ~~remember settings for website or maybe even allow creating pre sets that are automatically applied~~ (later)
- [x] after possible manual application of importer stuff dont show special importer page but send data to the default recipe editor (maybe just ask about step splitting and keyword importing on some kind of wizard page)
- [x] auto import list of urls
- [x] automatically tag based on website url (only first and second level domain)

todos

- [x] verify all places that use the ingredient parser/ source parser use the new field names/structure
- [x] ~~rate limit api endpoint for import~~ (later)
- [x] rewrite tests to fit new structure
- [x] fix bookmarklet flow again
- [x] fix recipe from source import

UI
- [x] improve export view (explain "all" checkbox, make it more intuitive)
- [x] improve importer/exporter selection (maybe logos, icons, nice carousel ...)
- [x] show original text in recipe editor
- [x] show source url field somewhere
- [x] localize missing fields
- [x] make button "single/multiple" nicer
- [x] remove recent imports list when finished

**Comment by choss** [comment]
I've got a question regarding this issue:
Can you make the export functionality also available via api? 
I am looking to export all recipes in tandoorRecipes format, so I can use them for my backup (instead of backing up the images and the postgresql)

**Comment by DiversityBug** [comment]
Hi Ben @vabene1111 
We have used your excellent work for years; the family loves it.
I had to upgrade the stack from an old Postgres 11, which fixed the search issue of finding nothing anymore.
Since Postgres 11 and 15 aren't compatible to reference in the new setup, I had to opt for exporting and importing the recipes.
Unfortunately, all nutritional values, logged cooking, and comments are lost in the process. (Same in your hosted demo).
Must I assume a technical limitation or is there something to do about it?
Cheers

**Comment by smilerz** [comment]
In order to upgrade postgress you need to backup the postgres db (v11) and import it into the upgraded postrgres (v15)

**Comment by DiversityBug** [comment]
Thanks for the quick reply @smilerz.
I assume Google is my friend or are there some instructions you could recommend (for Synology Docker users)?

**Comment by smilerz** [comment]
> Thanks for the quick reply @smilerz. I assume Google is my friend or are there some instructions you could recommend (for Synology Docker users)?

There might be instructions in the issues somewhere - but I don't know off the top of my head.

**Comment by DiversityBug** [comment]
Hi there,
I looked at it but couldn't make it click.
I spun up a pgadmin docker, connected to the old and new db. I got the required roles and a dump into the new. Everything is looking fine after copying the media files, except for all the books being lost.
Though, I'm testing a Portainer Stack for Tandoor on a Synology deployment via script and docker-compose which will take a backup as well... and probably include pgadmin. I'll share some documentation once done.

I posted this here since **exporting and importing from and to the same source Tandoor is leaving data behind. Is there a particular reason?** (Could be helpful to move recipes between spaces or from self-hosting to the new cloud offering.)

On a new topic.
Turns out the zero search results for a user or new admin are back. Only the space creator is turning up results. 
A bit of testing shows one needs to leave and be invited again to the space the admin created (before there were spaces).
Is there a "cleaner" way to fix that behavior?
Cheers
